zebra/zebra_vty.c

```diff
@@ -292,7 +491,33 @@ static int zebra_static_route(struct vty *vty, afi_t afi, safi_t safi,
 				       distance_str, label_str);
 }
 
+void static_config_install_delayed_routes(struct zebra_vrf *zvrf)
+{
+	struct listnode *node, *nnode;
+	struct static_hold_route *shr;
+	struct zebra_vrf *ozvrf, *nh_zvrf;
+
+	for (ALL_LIST_ELEMENTS(static_list, node, nnode, shr)) {
+		ozvrf = zebra_vrf_lookup_by_name(shr->vrf_name);
+		nh_zvrf = zebra_vrf_lookup_by_name(shr->nhvrf_name);
 
+		if (ozvrf != zvrf && nh_zvrf != zvrf)
+			continue;
+
+		if (ozvrf->vrf->vrf_id == VRF_UNKNOWN
+		    || nh_zvrf->vrf->vrf_id == VRF_UNKNOWN)
+			continue;
+
+		zebra_static_route_leak(
+			NULL, ozvrf, nh_zvrf, shr->afi, shr->safi, NULL,
+			shr->dest_str, shr->mask_str, shr->src_str,
+			shr->gate_str, shr->ifname, shr->flag_str, shr->tag_str,
+			shr->distance_str, shr->label_str);
+
```

Comment:
getting the returned value of zebra_statis_route_leak could permit to know exactly if an entry has been installed or not ( case mpls is not present in kernel)

Reply:
Not much to do with it other than to notice the rejection.